### ISSUE TITLE
Push Releases to rubygems.org

### ISSUE THREAD

**Issue by hh** [hh/f5-icontrol#3]
It was the most requested 'feature' in the comments at https://devcentral.f5.com/d/icontrol-ruby-library

Let's try and get this done when the gem works with current ruby releases (2.1.x) and current software relases (with SSL fixes #1 and #2 ).

**Comment by hh** [comment]
@jeffsilverm @grahamb  @jthurman42 @swalberg @jessereynolds You were each listed in the comments, so I thought I'd see if would be interested in joining the conversation.

**Comment by hh** [comment]
@swalberg I just noticed https://github.com/swalberg/f5-icontrol nice work!

**Comment by swalberg** [comment]
Thanks @hh, if you're using the original F5 one it's a pretty straightforward conversion and (IMHO) is more idiomatic Ruby.

**Comment by hh** [comment]
@joepruitt @onryo getting pushback from F5 regarding this gem being community only and not supported outside the devcentral community site. Do either of you have any cycles to help get it working (and maybe released to rubygems?)

**Comment by hh** [comment]
@swalberg have any thoughts on if we should try and just push your version as the rubygems.org version or this one? Are you using it currently / does it work? :)

**Comment by hh** [comment]
I opened F5Networks/f5-icontrol-rest-python#68 to try and get some visibility with folks currently working on f5-icontrol, even if it's not ruby (and not the wsdl version that this gem contains).

**Comment by hh** [comment]
Also opened https://devcentral.f5.com/questions/icontrol-rest-or-wsdl-rubygemsorg-release

**Comment by swalberg** [comment]
@hh yes I use mine constantly and also have a chef provider we use around it. About a year ago some guy at F5 reached out to me about using my code but that never went anywhere. I'll follow up.

SOAP vs REST - we were behind the times on upgrading to something that supported REST which is why I went down the SOAP route. What I'd like is for the gem to support both by providing a raw interface to the APIs, and then some wrappers to do common functions. Some of that is already done but it's sitting in the Chef provider rather than this gem.

**Comment by onryo** [comment]
@hh @swalberg Hey guys, thanks for reaching out. Unfortunately, I left F5 two years ago and haven't done any work on the iControl Ruby Gems since. I would reach out to @joepruitt or Jason on the DevCentral team.

**Comment by swalberg** [comment]
Just for some colour, my original intent was to modernize the F5 gem just like @hh and get it up on RubyGems.org. I ended up re-writing it because there's not a lot to it (it's a thin abstraction against the SOAP API), and with a rewrite I could make it more enjoyable to use.

So instead of stuff like `client['LocalLB.Pool'].get_member_v2(pool_names)` (and that's after loading `LocalLB.Pool` as an interface) you get `client.LocalLB.Pool.get_member_v2(pool_names)`. And, as a bonus, there's a CLI tool that comes with the gem so you can do `f5 pool list` and `f5 pool status POOL`.

Give it a shot - `gem install f5-icontrol` and check out the docs at  https://github.com/swalberg/f5-icontrol . Always happy to take PRs. And if you use Chef, there's also https://github.com/swalberg/chef-f5 which we use to create pools and VIPs (there's also the one from Target which is far more complete, but has a fairly different model for how it works)

**Comment by hh** [comment]
Nice, a couple chef cookbooks to choose from: https://github.com/target/f5-bigip-cookbook and yours!
Target went with the 'semiofficial' release https://github.com/target/f5-bigip-cookbook/blob/master/files/default/f5-icontrol-11.4.1.0.gem
I'll take a look sometime this week.

**Comment by hh** [comment]
Looks like f5-icontrol was already pushed by @swalberg with TOTAL DOWNLOADS
5,486 :) https://rubygems.org/gems/f5-icontrol/versions/0.1.5

**Comment by hh** [comment]
@jmccann what versions of ruby / ssl are you using in production with the devcentral gem release? It might be nice to know if the recent CVE / SSL updates to BigIP had any impact on the API endpoint.

**Comment by joepruitt** [comment]
All, I've just created the repo under F5's account https://github.com/f5networks/f5-icontrol-library-ruby from the DevCentral source.  This was developed by an ex employee so there currently is no internal support for it but any contributions you all make, I can review and get pushed in.  My first step is to get the 12.0.0 WSDL files incorporated.  I'll look into what @swalberg has done and would love to get those features put in this as well.  I haven't done much with Ruby so would appreciate as much help as you all are willing to give.

**Comment by hh** [comment]
Excellent @joepruitt ! Let me know when you get the 12.0.0 wsdl incorporated and I'll give it a spin.

**Comment by caphrim007** [comment]
@hh and @swalberg if you guys have stuff you want to send PRs on the f5networks github repo that @joepruitt mentioned, that I can probably build releases and send them to rubygems. I do that already with pypi and bigsuds

**Comment by swalberg** [comment]
Thanks @caphrim007... My gem is completely different though. Different API and different name on rubygems.org. Anything I submit would break people that already rely on the old F5 version.

The build stuff isn't a big deal as cutting a new release and uploading to rubygems is just one `rake` task.

A few options that I can see:
- the community works on `f5-icontrol-library-ruby`and we have two gems out there.
- F5 points people to the `f5-icontrol` that's already on Rubygems.org and lets the community work on the version I made
- I transfer my code to the F5 organization and it becomes the official version 12.0.0.

**Comment by caphrim007** [comment]
ahhhh @swalberg I see.
Well I'm not familiar with who uses the original sources, so if you want to be the official package, that's fine with me. afaik, all the code that goes on github and devcentral is more-or-less "f5 contributed, not supported". So the only thing that would make your release more official, so-to-speak, is if it were under the f5networks github account.
The best I can offer is testing if whatever y'all produce, because I've got access to more f5 devices than you can shake a stick at.

**Comment by jmccann** [comment]
@hh The bigip cookbook uses the version of ruby embedded with Chef, so it seems to be working at least with 2.1.6p336.
